src/canvas/Renderer.tsx

```diff
@@ -89,12 +89,16 @@ export class Renderer implements IRenderer {
 
                     const pathCrossedStartPoint = intersection.pathCrossed.getPoints()[0];
                     const pathCrossedEndpoint = intersection.pathCrossed.getPoints()[intersection.pathCrossed.getPoints().length - 1];
-                    if (intersection.point.isWithinRadius(pathCrossedEndpoint, 10) || intersection.point.isWithinRadius(pathCrossedStartPoint, 10)) {
-                        this._endTracing(intersection.point);
+                    if (intersection.point.isWithinRadius(pathCrossedEndpoint, 10)) {
```

Comment:
What if instead of cloning here, getPoints returned a list of cloned points?

Reply:
it would mean cloning 100 point in order to only use 2. What if we added a getStart() and getEnd() methods to PatternPath that would return a clone ? this way these accessors would be safe to use

Reply:
getStart and getEnd accesors in patternpath will get super useful too when a path has many segments

Reply:
and yeah I'll also have getPoints return clones, good idea

Reply:
Yeah, I think getStart and getEnd are a really good idea

Reply:
I pushed changes. Let me know if you like them.